packages/smooth_app/lib/database/dao_product.dart

```diff
@@ -130,4 +154,29 @@ class DaoProduct extends AbstractSqlDao implements BulkDeletable {
         json.decode(encodedJson) as Map<String, dynamic>;
     return Product.fromJson(decodedJson);
   }
+
+  Future<String> getFreshLocalDataBase() async {
+    try {
+      final List<String> barcodes = await getAllKeys();
+      if (barcodes.isEmpty) {
+        return 'List is Empty\nNothing to Refresh';
+      }
+      final ProductListQueryConfiguration configuration =
+          ProductListQueryConfiguration(
+        barcodes,
+        fields: ProductQuery.fields,
+        language: ProductQuery.getLanguage(),
+        country: ProductQuery.getCountry(),
+      );
+      final User user = ProductQuery.getUser();
+      // TODO(ashaman999): find a better way to do this not sure at what length it will break
+      final SearchResult products =
+          await OpenFoodAPIClient.getProductList(user, configuration);
+      final List<Product>? productList = products.products;
+      await putAll(productList!);
+      return 'OK';
+    } catch (e) {
+      return 'Error: $e';
+    }
```

Comment:
I would rather return nothing (or 0 and 1) and catch the error where we call the method

Reply:
Was better to show the errors to the user, should I modify it for sure?